expertise/models/multifacet_recommender/specter.py

```diff
@@ -81,6 +87,22 @@ def run(self) -> None:
 
         if self._output_file is not None:
             self._output_file.close()
+        self._redis_con.close()
+
+    def _maybe_print_to_console_and_file(self,
```

Comment:
who's calling this function?

Reply:
It's called by specter internally while calculating embeddings.

Reply:
I see, it is a function defined by specter.